Feature/filters

talorlangnas-moveo opened the pull request.

talorlangnas-moveo committed:
Add Footer component to RootLayout and replace it with AboutUs in Home page

- Integrated Footer component into RootLayout for consistent site structure.

- Removed Footer from Home page and added AboutUs component instead.

- Cleaned up Footer component by removing unnecessary AboutUs import.

talorlangnas-moveo committed:
refactoring header and footer file names to tmp names

talorlangnas-moveo committed:
refactoring header and footer file names to original names

talorlangnas-moveo committed:
Merge branch 'development' into feature/restaurants

talorlangnas-moveo committed:
Refactor restaurant components and data handling

- Simplified Layout component structure.

- Updated Restaurants component to fetch and display restaurant data using new utility functions.

- Modified Card component to comment out unused props.

- Corrected image URLs in restaurants.json to remove unnecessary prefixes.

- Enhanced fetchCards utility with new functions for fetching restaurant data and converting it to card format.

talorlangnas-moveo committed:
none

talorlangnas-moveo committed:
modifying restaurant data

talorlangnas-moveo committed:
RestaurantsHome component is ready for mobile

talorlangnas-moveo committed:
Merge branch 'development' into feature/restaurants

talorlangnas-moveo committed:
Refactor restaurant data fetching and layout adjustments

- Updated the Home component to fetch restaurant cards using a new function.

- Removed the unused Layout component from the restaurants directory.

- Enhanced the Footer and Header styles with additional borders.

- Limited the number of displayed cards in the CardsDisplay component to the top three.

- Added responsive design features to the RestaurantsHome component, including a filter section for desktop views.

- Deleted the outdated restaurantsCards data file and adjusted related fetch functions.

talorlangnas-moveo committed:
Remove console log from RestaurantsHome component

talorlangnas-moveo committed:
Enhance RestaurantsHome component with filtering functionality and style updates

- Implemented state management for active filter selection in the RestaurantsHome component.

- Added filter functions to display restaurants based on criteria: All, New, Most Popular, and Open Now.

- Updated styles for text items and introduced a new .textBarItem class for better UI interaction.

- Adjusted card rendering to use filtered restaurant data instead of the original list.

Still encountering responsiveness issues

talorlangnas-moveo committed:
Merge branch 'feature/restaurants' into feature/filters

talorlangnas-moveo committed:
Enhance Card and RestaurantsHome components with style and functionality updates

- Added a className prop to the Card component for better customization.

- Updated card rendering in RestaurantsHome to apply the new className for restaurant cards.

- Cleaned up styles in card.module.scss and restaurantsHome.module.scss for improved layout and responsiveness.

Responsiveness issue has been fixed

taluliel18 commented on epicure-app/src/components/restaurantsHome/restaurantsHome.tsx:
```diff
@@ -0,0 +1,129 @@
+}: RestaurantsHomeProps) {
+  const isDesktopView = useIsDesktopView();
+  const [activeItem, setActiveItem] = useState("All");
+  const [fillteredRestaurants, setFillteredRestaurants] =
```
Change fillteredRestaurants to filteredRestaurants

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/restaurantsHome/restaurantsHome.tsx:
```diff
@@ -0,0 +1,129 @@
+              Map View
+            </p>
+          )}
+        </div>
```
You can simplify and clean up the button rendering inside barContainer by using an array of filter options and mapping over them instead of repeating <p> elements manually. This makes the code more maintainable and easier to extend.

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/restaurantsHome/restaurantsHome.tsx:
```diff
@@ -0,0 +1,129 @@
+            <p className={styles.text}>Rating</p>
+            <Image src={DownArrow} alt="Down Arrow" width={24} height={24} />
+          </div>
+        </div>
```
The filterContainer section can be optimized by mapping over an array of tab names (e.g., "Price Range", "Distance", "Rating") instead of repeating the same JSX three times. 
Why are you using a different style for the "Rating" tab compared to the others?

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/card/card.tsx:
```diff
@@ -26,11 +27,16 @@ export default function Card({
 
   return (
-    <div className={`${styles.cardContainer} ${styles[type || "dish"]}`}>
+    <div
+      className={`${styles.cardContainer} ${styles[type || "restaurant"]} ${
+        className || ""
+      }`}
+    >
```
Replaced string concatenation in the className with clsx, which was already installed in the project but not used here. This improves readability and prevents potential styling issues.

talorlangnas-moveo replied:
done

talorlangnas-moveo committed:
Refactor Card and RestaurantsHome components for improved functionality and styling

- Introduced clsx for conditional className handling in Card and RestaurantsHome components, enhancing style management.

- Updated filtering logic in RestaurantsHome to utilize a centralized filterOptions array, streamlining the filter implementation.

- Corrected spelling of 'filteredRestaurants' for consistency and clarity.
